src/pynxtools/dataconverter/nexus_tree.py

```diff
@@ -214,6 +215,121 @@ def __init__(
+                elem_enum_open = elem_enum.attrib.get("open", "false")
+
+                if elem_enum_open == "true":
+                    return True
+                            
+                elem_enum_items = []
+                for items in elem_enum.findall(f"nx:item", namespaces=namespaces):
+                    value = items.attrib["value"]
+                    if value[0] == "[" and value[-1] == "]":
+                        import ast
+
+                        try:
+                            elem_enum_items.append(ast.literal_eval(value))
+                        except (ValueError, SyntaxError):
+                            raise Exception(
+                                f"Error parsing enumeration item in the provided NXDL: {value}"
+                            )
+                    else:
+                        elem_enum_items.append(value)
+                
+                def convert_to_hashable(item):
+                    """Convert lists to tuples for hashable types, leave non-list items as they are."""
+                    if isinstance(item, list):
+                        return tuple(item)  # Convert sublists to tuples
+                    return item  # Non-list items remain as they are
+
+                set_items = {convert_to_hashable(sublist) for sublist in self.items}
+                set_elem_enum_items = {convert_to_hashable(sublist) for sublist in elem_enum_items}
+               
+                if not set(set_items).issubset(set_elem_enum_items):
+                    # Should we really be this strict here? Or can appdefs define additional terms?
+                    return False
+            return True
+        
+        def _check_dimensions_fit(xml_elem: ET._Element) -> bool:
+            if not self.shape:
+                return True
+            elem_dimensions = xml_elem.find(f"nx:dimensions", namespaces=namespaces)
+            if elem_dimensions is not None:
+                rank = elem_dimensions.attrib.get("rank")
+                if rank is not None and not isinstance(rank, int):
+                    try:
+                        int(rank)
+                    except ValueError:
+                        # TODO: Handling of symbols
+                        return True
+                elem_dim = elem_dimensions.findall("nx:dim", namespaces=namespaces)
+                elem_dimension_rank = rank if rank is not None else len(rank)
+                dims: List[Optional[int]] = [None] * int(rank)
+                for dim in elem_dim:
+                    idx = int(dim.attrib["index"])
+                    if "value" not in dim.attrib:
+                        # This is probably an old dim element with ref
+                        pass
+                    try:
+                        value = int(dim.attrib["value"])
+                        dims[idx - 1] = value
+                    except ValueError:
+                        # TODO: Handling of symbols
+                        pass
```

Comment:
Index would start with `0` if not, that is issue from app-def.
So, here we can think something like:
```suggestion
                for dim in elem_dim:
                    idx = int(dim.attrib["index"])
                    if (value := dim.attrib.get("value", None)):
                        # This is probably an old dim element with ref
                        try:
                            value = int(value)
                            dims[idx] = value
                         except ValueError:
                        # TODO: Handling of symbols
                             pass
                elem_shape = tuple(dims)
```

Reply:
Makes sense. I couldn't quite accept your code because it wasn't matching correctly formatting-wise, but I copied it and made your changes.